cmd/global-query/pkg/distributed/query.go

```diff
@@ -69,19 +58,11 @@ func (q *QueryRunner) Run(ctx context.Context, args *query.Args) (*results.Resul
 	// log the query
 	logger := logging.Logger().With("hosts", hostList)
 
-	// query pipeline setup
-	// sets up a fan-out, fan-in query processing pipeline
-	numRunners := len(hostList)
-	if q.maxConcurrent > 0 && q.maxConcurrent < numRunners {
-		numRunners = q.maxConcurrent
-	}
+	logger.Info("reading query results from querier")
 
-	logger.With("runners", numRunners).Info("dispatching queries")
 
 	finalResult := aggregateResults(ctx, stmt,
-		runQueries(ctx, numRunners,
-			prepareQueries(ctx, q.querier, hostList, &queryArgs),
-		),
+		q.querier.Query(ctx, hostList, &queryArgs),
```

Comment:
Nice. So for raw queries I'll just use a different pipeline instead of `aggregateResults`. Looks very clean! :partying_face:

Reply:
Exactly. Something like `printStream` or whatever.

Reply:
Sweet! Will look into that in the aftermath!